bittide/src/Bittide/SwitchDemoProcessingElement.hs

```diff
@@ -217,3 +221,83 @@ switchDemoPeWb SNat localCounter = Circuit go
         <$> writeVec
 
     (writeVec, wbS2M) = unbundle $ wbToVec <$> bundle readVec <*> wbM2S
+
+  mm =
+    MemoryMap
+      { tree = DeviceInstance locCaller "SwitchDemoPE"
+      , deviceDefs = deviceSingleton deviceDef
+      }
+  deviceDef =
```

Comment:
This function now both defines the hardware and the memory map. I'm assuming it's future work to define the hardware in a way that generates the memory map. Is that correct? If so, could you add that to the "This PR does NOT add:" list in the PR description?

Reply:
Yes, at the moment I just added the memory maps separately to the implementation. There's already building blocks that build it up automatically, like `singleMasterInterconnect` or `processingElement`, but a lot of different circuits use different ways to implement them, so there wasn't a single place I could have added a memory map implementation to help build up those device definitions :cry: Maybe one day though!

Reply:
Understandable. I have to build some components from the ground up soon, we can try to add new primitives then and over time phase out the non-MM ones.